### ISSUE TITLE
BUG: link check fails to detect sysroot on PPC if build_platform sysroot >2.12

### ISSUE THREAD

**Issue by h-vetinari** [conda/conda-build#5403]
### Checklist

- [X] I added a descriptive title
- [X] I searched open reports and couldn't find a duplicate

### What happened?

Many maintainers in conda-forge try to pay attention to the output of the link check, and fix eventual issues. Some feedstocks even turn any failures for overlinking/overdepending to errors; for example, matplotlib [is](https://github.com/conda-forge/matplotlib-feedstock/blob/57c44c24dabd496cde03837256e45616b17b2135/conda-forge.yml#L9-L10) such a feedstock.

With the long-coming [switch](https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/6070) from cos6 -> cos7 (and thus bumping the glibc/sysroot version from 2.12 to 2.17), we got rid of the track-feature used to weigh down newer sysroots (that we neede vis-à-vis the 2.12 baseline) in https://github.com/conda-forge/linux-sysroot-feedstock/pull/65.

This had the curious effect that https://github.com/conda-forge/matplotlib-feedstock/pull/390 first ran with the following build environment:

```
    sysroot_linux-64:                          2.12-he073ed8_17           conda-forge
    sysroot_linux-ppc64le:                     2.17-h23d7e6c_14           conda-forge
```
leading to a clean link-check
```
   INFO (matplotlib-base,lib/python3.9/site-packages/matplotlib/_image.cpython-39-powerpc64le-linux-gnu.so): Needed DSO lib/libstdc++.so.6 found in conda-forge/linux-ppc64le::libstdcxx-ng==14.1.0=h262982c_0
   INFO (matplotlib-base,lib/python3.9/site-packages/matplotlib/_image.cpython-39-powerpc64le-linux-gnu.so): Needed DSO powerpc64le-conda-linux-gnu/sysroot/lib64/libm.so.6 found in CDT/compiler package conda-forge/noarch::sysroot_linux-ppc64le==2.17=h23d7e6c_14
   INFO (matplotlib-base,lib/python3.9/site-packages/matplotlib/_image.cpython-39-powerpc64le-linux-gnu.so): Needed DSO lib/libgcc_s.so.1 found in conda-forge/linux-ppc64le::libgcc-ng==14.1.0=h31e42bb_0
   INFO (matplotlib-base,lib/python3.9/site-packages/matplotlib/_image.cpython-39-powerpc64le-linux-gnu.so): Needed DSO powerpc64le-conda-linux-gnu/sysroot/lib64/libc.so.6 found in CDT/compiler package conda-forge/noarch::sysroot_linux-ppc64le==2.17=h23d7e6c_14
   INFO (matplotlib-base,lib/python3.9/site-packages/matplotlib/_image.cpython-39-powerpc64le-linux-gnu.so): Needed DSO powerpc64le-conda-linux-gnu/sysroot/lib64/ld64.so.2 found in CDT/compiler package conda-forge/noarch::sysroot_linux-ppc64le==2.17=h23d7e6c_14
                                                                                                                                                                  ^^^^^^^^^
                                                                                                                                                                  see here
```
while the CI run on main (after https://github.com/conda-forge/linux-sysroot-feedstock/pull/65) pulled in the following
```
    sysroot_linux-64:                          2.17-h4a8ded7_16           conda-forge
    sysroot_linux-ppc64le:                     2.17-h23d7e6c_16           conda-forge
```
and produced
```
   INFO (matplotlib-base,lib/python3.9/site-packages/matplotlib/_image.cpython-39-powerpc64le-linux-gnu.so): Needed DSO lib/libstdc++.so.6 found in conda-forge/linux-ppc64le::libstdcxx-ng==14.1.0=h262982c_0
   INFO (matplotlib-base,lib/python3.9/site-packages/matplotlib/_image.cpython-39-powerpc64le-linux-gnu.so): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libm.so.6 found in CDT/compiler package conda-forge/noarch::sysroot_linux-64==2.17=h4a8ded7_16
   INFO (matplotlib-base,lib/python3.9/site-packages/matplotlib/_image.cpython-39-powerpc64le-linux-gnu.so): Needed DSO lib/libgcc_s.so.1 found in conda-forge/linux-ppc64le::libgcc-ng==14.1.0=h31e42bb_0
   INFO (matplotlib-base,lib/python3.9/site-packages/matplotlib/_image.cpython-39-powerpc64le-linux-gnu.so): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libc.so.6 found in CDT/compiler package conda-forge/noarch::sysroot_linux-64==2.17=h4a8ded7_16
  ERROR (matplotlib-base,lib/python3.9/site-packages/matplotlib/_image.cpython-39-powerpc64le-linux-gnu.so): $RPATH/ld64.so.2 not found in packages, sysroot(s) nor the missing_dso_whitelist.
.. is this binary repackaging?
```

To sum up my understanding:
* This only happens with ppc64le, not with aarch64 (which is cross-compiled in exactly the same way)
* This doesn't affect the entirety of the sysroot [contents](https://conda-metadata-app.streamlit.app/?q=conda-forge%2Fnoarch%2Fsysroot_linux-ppc64le-2.17-h23d7e6c_16.conda), only `ld64.so.2` it seems
* However, the existence/location of `ld64.so.2` in `sysroot_linux-ppc64le` hasn't changed from `_14` to `_16`
* Moreover it doesn't trigger if the sysroot of the _build_ environment is 2.12...?!

Xref: https://github.com/conda-forge/linux-sysroot-feedstock/issues/67

### Conda Info

```shell
conda-forge CI
```

### Conda Config

```shell
conda-forge CI
```

### Conda list

```shell
conda-forge CI
```

**Comment by h-vetinari** [comment]
In terms of work-arounds (short of removing `error_overlinking: true` and living with the warnings):

`overlinking_ignore_patterns`: specific ❌ 

```
      overlinking_ignore_patterns:  # [ppc64le]
        - $RPATH/ld64.so.2          # [ppc64le]
```

`overlinking_ignore_patterns`: wildcard ✅ 

```
      overlinking_ignore_patterns:  # [ppc64le]
        - "**"                      # [ppc64le]
```

`missing_dso_whitelist`: specific ✅ 

```
      missing_dso_whitelist:    # [ppc64le]
        - $RPATH/ld64.so.2      # [ppc64le]
 ```

So it looks like there's some inconsistencies hiding in `overlinking_ignore_patterns` as well (at least w.r.t. the less obscure `missing_dso_whitelist`).

**Comment by isuruf** [comment]
It's not only `ld64.so.2` that's misidentified. There's `Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libc.so.6` which is the wrong sysroot to look at.

Looking at it more

Failing builds have
```
INFO: sysroot: '$BUILD_PREFIX/x86_64-conda-linux-gnu/sysroot/' files: '['usr/share/zoneinfo', 'usr/share/locale/zh_TW/LC_MESSAGES/libc.mo', 'usr/share/locale/zh_CN/LC_MESSAGES/libc.mo', 'usr/share/locale/vi/LC_MESSAGES/libc.mo']'
INFO: sysroot: ''$BUILD_PREFIX/powerpc64le-conda-linux-gnu/sysroot/' files: '['usr/share/zoneinfo', 'usr/share/locale/zh_TW/LC_MESSAGES/libc.mo', 'usr/share/locale/zh_CN/LC_MESSAGES/libc.mo', 'usr/share/locale/vi/LC_MESSAGES/libc.mo']'
```
while previously passing builds had
```
INFO: sysroot: ''$BUILD_PREFIXpowerpc64le-conda-linux-gnu/sysroot/' files: '['usr/share/zoneinfo/zone1970.tab', 'usr/share/zoneinfo/zone.tab', 'usr/share/zoneinfo/tzdata.zi', 'usr/share/zoneinfo/right/Zulu']'
INFO: sysroot: ''$BUILD_PREFIX/x86_64-conda-linux-gnu/sysroot/' files: '['usr/share/zoneinfo/zone.tab', 'usr/share/zoneinfo/tzdata.zi', 'usr/share/zoneinfo/right/Zulu', 'usr/share/zoneinfo/right/WET']'
```

i.e. order matters.

This reminded me of a core call where someone (@mbargull ? @beckermr ?) mentioned that the number of files in the sysroot is taken in for ordering. See https://github.com/conda/conda-build/blob/0c136f5000bed769924faefdbbc045b8ffe90218/conda_build/post.py#L1364-L1365

**Comment by h-vetinari** [comment]
I tested #5404 in matplotlib, and the sysroot is found correctly with it:
```
   INFO (matplotlib-base,lib/python3.9/site-packages/matplotlib/_image.cpython-39-powerpc64le-linux-gnu.so): Needed DSO lib/libstdc++.so.6 found in conda-forge/linux-ppc64le::libstdcxx-ng==14.1.0=h262982c_0
   INFO (matplotlib-base,lib/python3.9/site-packages/matplotlib/_image.cpython-39-powerpc64le-linux-gnu.so): Needed DSO powerpc64le-conda-linux-gnu/sysroot/lib64/libm.so.6 found in CDT/compiler package conda-forge/noarch::sysroot_linux-ppc64le==2.17=h23d7e6c_16
   INFO (matplotlib-base,lib/python3.9/site-packages/matplotlib/_image.cpython-39-powerpc64le-linux-gnu.so): Needed DSO lib/libgcc_s.so.1 found in conda-forge/linux-ppc64le::libgcc-ng==14.1.0=h31e42bb_0
   INFO (matplotlib-base,lib/python3.9/site-packages/matplotlib/_image.cpython-39-powerpc64le-linux-gnu.so): Needed DSO powerpc64le-conda-linux-gnu/sysroot/lib64/libc.so.6 found in CDT/compiler package conda-forge/noarch::sysroot_linux-ppc64le==2.17=h23d7e6c_16
   INFO (matplotlib-base,lib/python3.9/site-packages/matplotlib/_image.cpython-39-powerpc64le-linux-gnu.so): Needed DSO powerpc64le-conda-linux-gnu/sysroot/lib64/ld64.so.2 found in CDT/compiler package conda-forge/noarch::sysroot_linux-ppc64le==2.17=h23d7e6c_16
```

Thanks a lot Isuru! 🙏 

Given the importance of the sysroot[^1], I think this would be worth considering to still squeeze into 24.7, even though the hour is late. 😅 
CC @beeankha @kenodegard 

[^1]: this gets misdetected in basically every recipe, even though it only fails if `error_overlinking: true`
